iModelCore/ECDb/ECDb/ECSql/ECSqlPreparer.cpp

```diff
@@ -2781,6 +2838,104 @@ ECSqlStatus ECSqlExpPreparer::GenerateECClassIdFilter(Utf8StringR filterSqlExpre
     return ECSqlStatus::Success;
     }
 
+//-----------------------------------------------------------------------------------------
+// @bsimethod
+//+---------------+---------------+---------------+---------------+---------------+------
+//static
+ECSqlStatus ECSqlExpPreparer::PrepareNavValueCreationFuncExp(NativeSqlBuilder::List& nativeSqlSnippets, ECSqlPrepareContext& ctx, NavValueCreationFuncExp const& exp)
+    {
+    if (!ctx.GetECDb().Schemas().GetClass(exp.GetClassRefExp()->GetAs<ClassNameExp>().GetSchemaName(), exp.GetClassRefExp()->GetAs<ClassNameExp>().GetClassName())->GetPropertyP(exp.GetColumnRefExp()->GetAs<DerivedPropertyExp>().GetExpression()->GetAs<PropertyNameExp>().GetPropertyPath()[0].GetName())->GetIsNavigation())

```

Comment:
This looks like it can use some cleanup. It is long and hard to read and making several assumptions about the types of Exp objects which I'm not sure will always be correct.

Reply:
Refactored this place a bit